### ISSUE TITLE
Division by zero NaN vulnerability in categorical_crossentropy

### ISSUE THREAD

**Issue by Deeven-Seru** [keras-team/keras#23551]
### Description of the Vulnerability

There is a division-by-zero NaN vulnerability in the `categorical_crossentropy` loss function across all Keras 3 backends (Legacy, TensorFlow, PyTorch, JAX, NumPy, OpenVINO). 

When the model predictions (outputs) sum to exactly zero, the normalization step:
`output = output / reduce_sum(output, axis, keepdims=True)`
results in a division by zero. This immediately propagates `NaN`s through the loss and poisons the gradients, causing catastrophic training failure. This edge case is particularly prevalent when models are fine-tuned on resource-constrained hardware or with extreme learning rates where outputs might collapse to zero across the class dimension.

### Proposed Mitigation

This can be seamlessly mitigated by adding `backend.epsilon()` to the denominator prior to division, ensuring numerical stability without impacting the loss calculation for valid distributions:

```python
output = output / (reduce_sum(output, axis, keepdims=True) + backend.epsilon())
```

### Upstream Context

This issue was originally identified and patched in the TensorFlow repository, but the maintainers requested the fix be ported upstream to Keras to protect the wider ecosystem. 

- **Original TensorFlow PR**: https://github.com/tensorflow/tensorflow/pull/123494

I have opened a PR to address this across all Keras backends.

**Comment by dhantule** [comment]
Hi @Deeven-Seru, thanks for the fix. 

Attaching a reproducible [gist](https://colab.sandbox.google.com/gist/dhantule/488953676d8f0d985726ceef9a31f5e4/division-by-zero-nan-vulnerability-in-categorical_crossentropy-23551.ipynb).
